ml4gw/waveforms/generator.py

```diff
@@ -0,0 +1,39 @@
+from typing import Callable
```

Comment:
Part of me is wondering, given how simple this module is, if it makes sense to relegate this to downstream applications to implement as they see fit. Like to take BBHNet as an example, is using this simpler (or more clear) than going:

```
dec = self.dec(N)
psi = self.psi(N)
phi = self.phi(N)

idx = torch.randint(however_we_keep_track_of_num_waveforms, size=(N,))
polarizations = {k: v[idx] for k, v in self.polarizations.items()}
responses = self.projector(dec, psi, phi, **polarizations)
```

I do like the idea of the `ParameterSampler` as a sort of torch-ified extension of the bilby `PriorDict`, possibly adding `.prob` functionality down the line, but in that case it might make more sense as a `dict` subclass with a `.sample` method. The main benefit of torch `Module`s is for the tracking of parameters and the ability to export graphs of modules in one fell swoop, and I don't know that we need either of those benefits here.

Reply:
Yeah I agree both the `WaveformSampler` and `ParameterSampler` are quite thin wrappers at the moment. 
My vision was also an eventual bilby `PriorDict` like object with `.probs` etc. 

I'm really not sure what's best. I'm constantly debating what could benefit from generalization.

Reply:
I put together a [gist](https://gist.github.com/EthanMarx/919de0d738319581d0b97af42638f7ae) of what the PE projects injector would look like under the current implementation of the API to get a better sense of all the use cases.
 
I think one pro of the `ParameterSampler` is that it can abstract the sampling of many parameters in one object. This makes the PEInjector object agnostic to the waveform / parameters of the waveform. 

Then again, the `ParameterSampler` is something that could be implemented by downstream projects really easily as well... 

For the sake of getting the `SineGaussian` merged in happy to remove this module and continue this discussion in the injection refactoring PR.